fix(anthropic): use default thinking budget when unspecified

EurFelux opened the pull request:
## Background

When users attempt to use extended thinking with Anthropic models without giving `thinkingBudget`, they receive a weird error message `'thinking requires a budget' functionality not supported`.

## Summary

- use a default thinking budget of 1024 tokens
- return compatibility warning

## Manual Verification

- [x] ran `examples/ai-core/src/generate-text/anthropic-reasoning-without-budget.ts`

## Checklist

- [x] Tests have been added / updated (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] A _patch_ changeset for relevant packages has been added (for bug fixes / features - run `pnpm changeset` in the project root)
- [x] I have reviewed this pull request (self-review)

EurFelux committed:
fix(anthropic): improve error message for missing thinking budget

Make error message more specific when thinking is enabled but no budget is provided

EurFelux committed:
fix(anthropic): clarify error message for missing thinking budget

lgrammel commented:
please add a patch changeset https://github.com/vercel/ai/blob/main/CONTRIBUTING.md#submitting-pull-requests

lgrammel commented on packages/anthropic/src/anthropic-messages-language-model.ts:
```diff
@@ -371,7 +370,8 @@ export class AnthropicMessagesLanguageModel implements LanguageModelV3 {
-          functionality: 'thinking requires a budget',
+          functionality: 'extended thinking',
+          message: 'Extended thinking requires a thinking budget',
         });
```
ideally this should warn, not error

EurFelux committed:
fix(anthropic): handle missing thinking budget case gracefully

Instead of throwing an error when thinking is enabled without a budget, add a warning and proceed with default behavior. This provides better user experience by allowing the model to continue running with adjusted max tokens.

EurFelux committed:
fix(anthropic): update changeset

lgrammel committed:
x

lgrammel committed:
t

lgrammel committed:
x
